client/dashboard/app/router.tsx

```diff
@@ -105,6 +106,17 @@ const sitesRoute = createRoute( {
 const siteRoute = createRoute( {
 	getParentRoute: () => rootRoute,
 	path: 'sites/$siteSlug',
+	beforeLoad: async ( { cause, params: { siteSlug }, location } ) => {
+		if ( cause !== 'enter' ) {
+			return;
+		}
+
+		const site = await queryClient.ensureQueryData( siteBySlugQuery( siteSlug ) );
+		const trialExpiredUrl = `/sites/${ siteSlug }/trial-ended`;
+		if ( hasSiteTrialEnded( site ) && ! location.pathname.includes( trialExpiredUrl ) ) {
+			throw redirect( { to: trialExpiredUrl } );
```

Comment:
I know we redirect in v1, but I'm wondering is redirecting the nicest solution here? Couldn't the site router just render a whole different component if the trial has expired?

Maybe redirecting is the easiest way to change the secondary nav though 🤷